### DISCUSSION TITLE
prefixed tailwindcss

### DISCUSSION THREAD

**Discussion by rch77** [mdbootstrap/TW-Elements#1946] [Share ideas & request features]
Hi there,
I think it misses support for the prefixed version of tailwindcss or I can't find it in the documentation. All examples place classes in common global namespace which obviously doesn't work if you have a project with some older existing css to merge.
I understand I can add tw- prefixes manually but it's too time consuming and error-prone as it is not obvious which complex classes/selectors should have the prefix where exactly.

**Comment by juujisai** [comment]
Hi, our components are made with use some of the newest features of tailwind. Some of the components won't work on the older versions.

**Comment by rch77** [comment]
I mean the `prefix` option.
https://tailwindcss.com/docs/configuration#prefix

**Comment by juujisai** [comment]
Oh I see. I'll add this idea to our list and we will keep that in mind

**Comment by thehands79** [comment]
Hi, was support ever added for prefixing?

**Reply by juujisai** [reply]
Hello. I am researching the possibilities for that at this very moment but the chances for us to do something about it are slim. Lots of our components are using tailwind classes inside the js files. Since tailwind doesn't support creating dynamic classes (for example if we were to add ${prefix}bg-red-600) there is not much we can do.

I'm still going to research what we can do about this but unfortunately at this moment the only thing I can see is to use our class customization and add prefixes manually.

**Reply by thehands79** [reply]
Hi, thanks for the info. Do you have a bit more detail on how "to use our class customization and add prefixes manually"? Is this less manual than writing some script to add the prefix into tw-elements.css which is one option I had considered. However it sounds like this won't work for components that use the js files you mentioned.

**Reply by juujisai** [reply]
The class customization can be achieved via data attributes or via JS. Example can be found in this issue:
https://github.com/mdbootstrap/TW-Elements/issues/2033

Our component pages contain API tabs where we included `Classes` subsection. You can find there elements that we add inside our JS files and classes that are used for them:
https://tw-elements.com/docs/standard/forms/datepicker/#docsTabsAPI
